pkg/config/config.go

```diff
@@ -82,6 +82,7 @@ type Config struct {
 	ConfigFolder                 string                `json:"configFolder"`
 	LogLevel                     string                `json:"logLevel"`
 	Logger                       map[string]string     `json:"logger"`
+	RemoteMining                 bool                  `json:remoteMining`
```

Comment:
Do you mean to have multiple config files or just command-specific fields in a single config file?

Reply:
We can keep a single file, but with some more structured format. Something like this would be a good start. But lets do it in a separate PR , for now just add `DataserverHost`, and `DataserverPort`

```
type Config struct{
	miner Mine type
	dataserver Dataserver type
	... other fields
}

{
	"contractAddress": "0xFe41Cb708CD98C5B20423433309E55b53F79134a",
    "publicAddress": "0x0e93f23278161bCa6F22D6268ca44691042ed437",
	"mine":{
		"host": "0.0.0.0", // It serves the metrics here. But later can expose some api.
    	"port": 5000,
		DataserverHost:... // Connect to this remote dataserver
		DataserverPort:...
		"dbFile": ".local/tellorDB", //  When DataserverHost is set this is not used.
	}
	"dataserver": {
		"host": "0.0.0.0",
    	"port": 5000,
		"dbFile": ".local/tellorDB",
	}
    
    "gasMax": 10,
    "configFolder": "configs",
    "profitThreshold": 50,
    "minConfidence": 0,
    "envFile": ".local/.env",
    "LogLevel": "debug",
    "Logger": {
        "config.Config": "INFO",
        "db.DB": "INFO",
        "rpc.client": "INFO",
        "rpc.ABICodec": "INFO",
        "rpc.mockClient": "INFO",
        "tracker.Top50Tracker": "INFO",
        "tracker.FetchDataTracker": "INFO",
        "pow.MiningWorker-0:": "INFO",
        "pow.MiningWorker-1:": "INFO",
        "pow.MiningTasker-0:": "INFO",
        "pow.MiningTasker-1:": "INFO",
        "tracker.PSRTracker": "INFO"
    }
}
```

Reply:
Currently, ServerHost and ServerPort are only used for the remote connections, the local connection doesn't use those. I'll just leave them as is, ok?